test/unit/visitor/sympy_solver.cpp

```diff
@@ -577,6 +614,214 @@ SCENARIO("Solve ODEs with derivimplicit method using SympySolverVisitor",
         }
     }
 
+    GIVEN("Derivative block, sparse, print in order") {
+        std::string nmodl_text = R"(
+            STATE {
+                x y
+            }
+            BREAKPOINT  {
+                SOLVE states METHOD sparse
+            }
+            DERIVATIVE states {
+                LOCAL a, b
+                y' = a
+                x' = b
+            })";
+        std::string expected_result = R"(
+            DERIVATIVE states {
+                LOCAL a, b, old_y, old_x
+                old_y = y
+                old_x = x
+                y = a*dt+old_y
+                x = b*dt+old_x
+            })";
+
+        THEN("Construct & solve linear system for backwards Euler") {
+            auto result =
+                run_sympy_solver_visitor(nmodl_text, false, false, AstNodeType::DERIVATIVE_BLOCK);
+
+            compare_blocks(reindent_text(result[0]), reindent_text(expected_result));
+        }
+    }
+    GIVEN("Derivative block, sparse, print in order, vectors") {
+        std::string nmodl_text = R"(
+            STATE {
+                M[2]
+            }
+            BREAKPOINT  {
+                SOLVE states METHOD sparse
+            }
+            DERIVATIVE states {
+                LOCAL a, b
+                M'[1] = a
+                M'[0] = b
+            })";
+        std::string expected_result = R"(
+            DERIVATIVE states {
+                LOCAL a, b, old_M_1, old_M_0
+                old_M_1 = M[1]
+                old_M_0 = M[0]
+                M[1] = a*dt+old_M_1
+                M[0] = b*dt+old_M_0
+            })";
+
+        THEN("Construct & solve linear system for backwards Euler") {
+            auto result =
+                run_sympy_solver_visitor(nmodl_text, false, false, AstNodeType::DERIVATIVE_BLOCK);
+
+            compare_blocks(reindent_text(result[0]), reindent_text(expected_result));
+        }
+    }
+    GIVEN("Derivative block, sparse, derivatives mixed with local variable reassignment") {
+        std::string nmodl_text = R"(
+            STATE {
+                x y
+            }
+            BREAKPOINT  {
+                SOLVE states METHOD sparse
+            }
+            DERIVATIVE states {
+                LOCAL a, b
+                    x' = a
+                    b = b + 1
+                    y' = b
+            })";
+        std::string expected_result = R"(
+            DERIVATIVE states {
+                LOCAL a, b, old_x, old_y
+                old_x = x
+                old_y = y
+                x = a*dt+old_x
+                b = b+1
+                y = b*dt+old_y
```

Comment:
not really because they can be used in the solution that you are substituting. For example, if sympy returns with:

```
x = f(x,y)
y = g(x,y)
```

and you do not use old_x old_y you end up using the new x and the old y for the computation of y but the old x and y for the computation of x since when you do the check you analyze the initial unsolved set of equations. You could analyze both old and new set of equations. However, this is much more complicatede and error-prone than just snapshotting x value at the beginning of the cycle